BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -381,10 +385,10 @@ extension KeyringStore: BraveWalletKeyringServiceObserver {
         break
       }
       
-      if selectedAccount.coin.keyringId != keyringId {
+      let newKeyring = await keyringService.keyringInfo(keyringId)
+      if let newAccount = newKeyring.accountInfos.first {
         walletService.setSelectedCoin(coin)
```

Comment:
if you set a new coin here without setNetwork
`selectedAccountChanged` will not take care of changing network since the coin is already set to the correct one.

Reply:
`NetworkStore` observes selected account changed and updates the network. However, this did catch an issue where `isSwapSupported` was not being updated when a Solana account was created outside of the network picker. This was addressed in [bc8e9c3](https://github.com/brave/brave-ios/pull/5845/commits/bc8e9c3c3a5ae8c3c049a6bbe391d8d0842c162b)